File: reset_password.mjs
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
dotenv.config({ path: '.env.local' });

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

const supabase = createClient(supabaseUrl, supabaseKey);

async function resetPassword() {
  const emailSearch = 'bionic'; // we'll match loosely
  const { data: users, error: listError } = await supabase.auth.admin.listUsers();
  
  if (listError) {
    console.error('List error:', listError);
    return;
  }
  
  console.log('List of users found:', users.users.map(u => ({ id: u.id, email: u.email })));
  
  const targetUser = users.users.find(u => u.email && (u.email.includes('bx') || u.email.includes('bionic')));
  
  if (!targetUser) {
    console.log('User not found!');
    return;
  }
  
  console.log(`Found target user! Email: ${targetUser.email}, ID: ${targetUser.id}`);
  
  const newPassword = "LivingWord2026!";
  const { data, error } = await supabase.auth.admin.updateUserById(
    targetUser.id,
    { password: newPassword }
  );
  
  if (error) {
    console.error('Update error:', error);
  } else {
    console.log(`SUCCESS! Changed password for ${targetUser.email} to: ${newPassword}`);
  }
}

resetPassword();

File: reset_password.py
import os
import sys
import json
import urllib.request
from dotenv import load_dotenv

load_dotenv('.env.local')

supabase_url = os.getenv("SUPABASE_URL")
supabase_key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

headers = {
    "apikey": supabase_key,
    "Authorization": f"Bearer {supabase_key}",
    "Content-Type": "application/json"
}

# 1. Fetch users
req = urllib.request.Request(f"{supabase_url}/admin/users", headers=headers)
try:
    with urllib.request.urlopen(req) as response:
        users_data = json.loads(response.read().decode())
        users = users_data.get('users', [])
except Exception as e:
    print(f"Error fetching users: {e}")
    sys.exit(1)

target = None
print("Users in database:")
for u in users:
    email = u.get("email", "")
    print(f" - {email}")
    if "bx4" in email.lower() or "bion" in email.lower():
        target = u

if not target:
    print("User bx4 or bione not found!")
    sys.exit(1)

print(f"\nFound target: {target['email']} (ID: {target['id']})")

# 2. Update user
update_req = urllib.request.Request(
    f"{supabase_url}/admin/users/{target['id']}", 
    headers=headers, 
    method="PUT",
    data=json.dumps({"password": "LivingWord2026!"}).encode()
)
try:
    with urllib.request.urlopen(update_req) as res:
        print(f"SUCCESS! Changed password for {target['email']} to: LivingWord2026!")
except Exception as e:
    print(f"Error updating user: {e}")
